In [1]:
import pandas as pd
import numpy as np

In [43]:
df = pd.read_csv("data_pancake.csv")
df.head()

Unnamed: 0             name gender                address      mobile  \
0           0  Ut Huynh Nguyen   male                    NaN  0909278702   
1           1     Nhậtt Hoàngg    NaN                    NaN  0838728888   
2           3      Khoa Nón Lá    NaN                    NaN  0838728888   
3           4       Nguyễn Nam    NaN                    NaN  0788728282   
4           5       Ngânn Monn    NaN  Thành phố Hồ Chí Minh  0788728282   

  source_lead note          page_id  
0         BOT   OD  106048591142776  
1         BOT   OD  106048591142776  
2         BOT   OD  106048591142776  
3         BOT   OD  106048591142776  
4         BOT   OD  106048591142776

In [44]:
df.drop(columns=df.columns[0], inplace=True)

In [45]:
def change_page_name(page):
    if page == 106048591142776:
        page = "Đào tạo lập trình MCI Việt Nam Tại TP. HCM"
    elif page == 102791862223216:
        page = "Học viên công nghệ MCI"
    elif page == 101464274796063:
        page = "Học viện công nghệ MCI Quận 3 - HCM"
    elif page == 100775119097939:
        page = "Học viện công nghệ MCI Đống Đa - HN"
    elif page == 101260665268485:
        page = "Học viện đào tạo công nghệ MCI VN"
    return page
change_page_name(102791862223216)

'Học viên công nghệ MCI'

In [48]:
for idx, each in df["page_id"].items():
    df["page_id"][idx] = change_page_name(each) 

In [49]:
df.fillna("", inplace=True)

In [50]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1808 entries, 0 to 1807
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   name         1808 non-null   object
 1   gender       1808 non-null   object
 2   address      1808 non-null   object
 3   mobile       1808 non-null   object
 4   source_lead  1808 non-null   object
 5   note         1808 non-null   object
 6   page_id      1808 non-null   object
dtypes: object(7)
memory usage: 99.0+ KB


In [51]:
df.groupby("page_id").count()

name  gender  address  mobile  \
page_id                                                                     
Học viên công nghệ MCI                       590     590      590     590   
Học viện công nghệ MCI Quận 3 - HCM          756     756      756     756   
Học viện công nghệ MCI Đống Đa - HN          174     174      174     174   
Học viện đào tạo công nghệ MCI VN            277     277      277     277   
Đào tạo lập trình MCI Việt Nam Tại TP. HCM    11      11       11      11   

                                            source_lead  note  
page_id                                                        
Học viên công nghệ MCI                              590   590  
Học viện công nghệ MCI Quận 3 - HCM                 756   756  
Học viện công nghệ MCI Đống Đa - HN                 174   174  
Học viện đào tạo công nghệ MCI VN                   277   277  
Đào tạo lập trình MCI Việt Nam Tại TP. HCM           11    11

### Tổng cộng 1808
### Học viện công nghệ MCI - có 590 người - 590 - OD
### Học viện công nghệ MCI Quận 3 - HCM 491 - FB & 265 - OD
### Học viện công nghệ MCI Đống Đa - HN 174 - OD
### Học viện đào tạo công nghệ MCI VN - 166 - FB, 111 - OD
### Đào tạo lập trình MCI Việt Nam Tại TP. HCM  - 11 - OD


In [77]:
df[df["page_id"] == "Đào tạo lập trình MCI Việt Nam Tại TP. HCM"].groupby("note")["mobile"].count()


note
OD    11
Name: mobile, dtype: int64